NeuroIDAdvancedDevice/AdvancedDevice.swift

```diff
@@ -36,30 +40,32 @@ public class NeuroIDADV: NSObject, DeviceSignalService {
             }
         }
     }
-    
+
     public static func getAdvancedDeviceSignal(_ apiKey: String, completion: @escaping (Result<String, Error>) -> Void) {
-          NeuroIDADV.getAPIKey(apiKey) { result in
-              switch result {
-              case .success(let fAPiKey):
-                  NeuroIDADV.retryAPICall(apiKey: fAPiKey, maxRetries: 3, delay: 2) { result in
-                      switch result {
-                      case .success(let (value, _)):
-                          completion(.success(value))
-                      case .failure(let error):
-                          completion(.failure(error))
-                      }
-                  }
-              case .failure(let error):
-                  completion(.failure(error))
-              }
-          }
-      }
-
-    internal static func getAPIKey(
+        NeuroIDADV.getAPIKey(apiKey) { result in
+            switch result {
+            case .success(let fAPiKey):
+                NeuroIDADV.retryAPICall(apiKey: fAPiKey, maxRetries: 3, delay: 2) { result in
+                    switch result {
+                    case .success(let (value, _)):
+                        completion(.success(value))
+                    case .failure(let error):
+                        completion(.failure(error))
+                    }
+                }
+            case .failure(let error):
+                completion(.failure(error))
+            }
+        }
+    }
+
+    static func getAPIKey(
         _ apiKey: String,
+        clientID: String? = "",
+        linkedSiteID: String? = "",
         completion: @escaping (Result<String, Error>) -> Void
     ) {
-        let apiURL = URL(string: "https://receiver.neuroid.cloud/a/\(apiKey)")!
+        let apiURL = URL(string: "https://receiver.neuroid.cloud/a/\(apiKey)?clientId=\(clientID ?? "")&linkedSiteId=\(linkedSiteID ?? "")")!
```

Comment:
Have empty parameter values been accounted for by/tested against collector-service?

Reply:
Not yet, the work to update the collector side is captured in https://neuro-id.atlassian.net/browse/ENG-8343 which states the query params are optional